src/Strategy/Node/Pnpm/PnpmLock.hs

```diff
@@ -121,6 +160,18 @@ instance FromJSON ProjectMap where
       <$> obj .:? "dependencies" .!= mempty
       <*> obj .:? "devDependencies" .!= mempty
 
+newtype ProjectMapDepMetadata = ProjectMapDepMetadata
+  { version :: Text
+  }
+  deriving (Show, Eq, Ord)
+
+instance FromJSON ProjectMapDepMetadata where
+  -- This is v5 lock format
+  parseJSON (Yaml.String r) = pure $ ProjectMapDepMetadata r
+  -- This is v6 lock format
+  parseJSON (Yaml.Object obj) = ProjectMapDepMetadata <$> obj .: "version"
+  parseJSON _ = fail "Expected to contain string or object with 'version' field!"
```

Comment:
Please print the thing that failed to parse properly to simplify later debugging.

e.g.:
```suggestion
  parseJSON other = fail ("Invalid format; expected pure string or an object with a `version` field, got: " <> show other)
```